presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -178,6 +182,21 @@ public Builder withParent(Scope parent)
         {
             checkArgument(!this.parent.isPresent(), "parent is already set");
             this.parent = Optional.of(parent);
+            withParentNamedQueries(parent);
+            return this;
+        }
+
+        public Builder withOuterQueryParent(Scope parent)
+        {
+            checkArgument(!this.parent.isPresent(), "parent is already set");
```

Comment:
`withParent` does copy parent named queries too, while when setting outer scope we wan't to provide different named queries parent